Source/Core/Helpers.cs

```diff
@@ -263,18 +264,39 @@ public static void ExtraTraceInformation(CoreOptions options, string point)
       }
     }
 
+    private static readonly HashSet<string> UsedLogNames = new();
+    public static (string fileName, bool reused) GetLogFilename(string descriptiveName, string filename, bool allowReuse)
+    {
+      filename = SubstituteAtProc(descriptiveName, cce.NonNull(filename));
+      var curFilename = filename;
+      var reused = false;
+
+      lock (UsedLogNames) {
+        int n = 1;
+        while (!allowReuse && UsedLogNames.Contains(curFilename)) {
+          curFilename = filename + "." + n++;
+        }
+
+        if (UsedLogNames.Contains(curFilename)) {
+          reused = true;
+        }
+
+        UsedLogNames.Add(curFilename);
+      }
+
+      return (curFilename, reused);
+    }
+
     // Substitute @PROC@ in a filename with the given descName
-    public static string SubstituteAtPROC(string descName, string fileName)
+    private static string SubstituteAtProc(string descriptiveName, string filename)
```

Comment:
I wouldn't change the name of this method; it substitutes `@PROC`, not `@Proc`

Reply:
Rider warns about any names with consecutive capitals, but I'm not sure why so I'll undo this.